signer/src/transaction_signer.rs

```diff
@@ -639,42 +640,52 @@ where
             .await
     }
 
-    /// Reach out to the blocklist client and find out whether we can
-    /// accept the deposit given all of the input `scriptPubKey`s of the
-    /// transaction. If the block list client is not configured then we
-    /// always accept the deposit request.
+    /// Check whether this signer accepts the deposit request. This
+    /// involves:
+    ///
+    /// 1. Reach out to the blocklist client and find out whether we can
+    ///    accept the deposit given all the input `scriptPubKey`s of the
+    ///    transaction.
+    /// 2. Check if we are a part of the signing set associated with the
+    ///    public key locking the funds.
+    ///
+    /// If the block list client is not configured then the first check
+    /// always passes.
     #[tracing::instrument(skip(self))]
     pub async fn handle_pending_deposit_request(
         &mut self,
         request: model::DepositRequest,
         bitcoin_chain_tip: &model::BitcoinBlockHash,
     ) -> Result<(), Error> {
-        let bitcoin_network = bitcoin::Network::from(self.context.config().signer.network);
-        let params = bitcoin_network.params();
-        let addresses = request
-            .sender_script_pub_keys
-            .iter()
-            .map(|script_pubkey| {
-                bitcoin::Address::from_script(script_pubkey, params)
-                    .map_err(|err| Error::BitcoinAddressFromScript(err, request.outpoint()))
-            })
-            .collect::<Result<Vec<bitcoin::Address>, _>>()?;
-
-        let is_accepted = futures::stream::iter(&addresses)
-            .any(|address| async { self.can_accept(&address.to_string()).await })
-            .await;
+        let db = self.context.get_storage_mut();
+
+        let signer_public_key = self.signer_pub_key();
+        // Let's find out whether or not we can even sign for this deposit
+        // request. If we cannot then we do not even reach out to the
+        // blocklist client.
+        //
+        // We should have a record for the request because of where this
+        // function is in the code path.
+        let can_sign = db
+            .can_sign_deposit_tx(&request.txid, request.output_index, &signer_public_key)
+            .await?
+            .unwrap_or(false);
+
+        let is_accepted = can_sign && self.can_accept_deposit_request(&request).await?;
 
         let msg = message::SignerDepositDecision {
             txid: request.txid.into(),
             output_index: request.output_index,
             accepted: is_accepted,
+            can_sign,
         };
 
         let signer_decision = model::DepositSigner {
             txid: request.txid,
             output_index: request.output_index,
-            signer_pub_key: self.signer_pub_key(),
+            signer_pub_key: signer_public_key,
```

Comment:
Suggestion: Not related specifically to this line, but after reading through a number of files I'm starting to dislike `signers_public_keys` as we also have `signer_public_key` like here. I'd almost prefer that we called it `signer_set_public_key`. Verbose, but very clear... thoughts?

Reply:
Just this comment alone left me overly confused 😂

But yeah, naming for the different concepts which are explicit and clear is needed regardless, imo

Reply:
We spoke about this yesterday. Basically, we should stick to a common name for the aggregate key in the code base. I'll open a ticket for the refactor for the name change stuff.

Reply:
I opened https://github.com/stacks-network/sbtc/issues/729 for the name changes.